src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -73,38 +74,51 @@ protected override void OnNextCore(Block value)
                 return;
             }
 
+            ChainedBlock chainedBlock = this.chain.GetBlock(block.GetHash());
+            if (chainedBlock == null)
+            {
+                this.logger.LogTrace("(-)[REORG]");
+                return;
+            }
+
+            BlockPair blockPair = new BlockPair(block, chainedBlock);
+
             // ensure the block is written to disk before relaying
-            this.blockStoreLoop.AddToPending(value);
+            this.blockStoreLoop.AddToPending(blockPair);
 
             if (this.chainState.IsInitialBlockDownload)
             {
                 this.logger.LogTrace("(-)[IBD]");
                 return;
             }
 
-            uint256 blockHash = value.GetHash();
-            this.logger.LogTrace("Block hash is '{0}'.", blockHash);
-            this.blockHashesToAnnounce.TryAdd(blockHash, blockHash);
+            this.logger.LogTrace("Block hash is '{0}'.", chainedBlock.HashBlock);
```

Comment:
we can log the whole chainedBlock just after we obtain it - i.e. before new BlockPair